### PR TITLE
materialize: 0.10.0 -> 0.15.0

### PR DESCRIPTION
**Pull request by petrosagg**
###### Motivation for this change

Update materialize

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/sql/materialize/default.nix [hunk 1]
```diff
@@ -68,8 +68,12 @@ rustPlatform.buildRustPackage rec {
     "--skip test_client"
     "--skip test_client_errors"
     "--skip test_client_all_subjects"
+    "--skip test_client_subject_and_references"
     "--skip test_no_block"
     "--skip test_safe_mode"
+    # this test is broken on 0.15.0
+    # TODO: re-add it in a subsequent release
```
Comment (JohnAZoidberg) [inline pkgs/servers/sql/materialize/default.nix:75]: Upstream is working on a fix?
Do they have an issue to track it?

Comment (petrosagg) [reply pkgs/servers/sql/materialize/default.nix:75]: disclaimer: I'm working for materialize.

This is already fixed in a pending PR. Despite the test's name this doesn't actually test anything to do with threading, it tests whether the executable can parse the command line parameter for specifying the number of threads correctly, and passing a non-UTF8 value produces a different error than what the test expects.

tl;dr: this test can be safely ignored without user impact and will be removed in the next release

Comment (JohnAZoidberg) [reply pkgs/servers/sql/materialize/default.nix:75]: Great, thanks! Sounds good.

I trust that you tested it adequately then.

**Review by JohnAZoidberg — APPROVED** [review]
